libvips/foreign/spngsave.c

```diff
@@ -0,0 +1,914 @@
+/* save to spng
+ *
+ * 2/12/11
+ * 	- wrap a class around the spng writer
+ * 16/7/12
+ * 	- compression should be 0-9, not 1-10
+ * 20/6/18 [felixbuenemann]
+ * 	- support spng8 palette write with palette, colours, Q, dither
+ * 24/6/20
+ * 	- add @bitdepth, deprecate @colours
+ * 11/11/21
+ * 	- use libspng for save
+ */
+
+/*
+
+    This file is part of VIPS.
+    
+    VIPS is free software; you can redistribute it and/or modify
+    it under the terms of the GNU Lesser General Public License as published by
+    the Free Software Foundation; either version 2 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU Lesser General Public License for more details.
+
+    You should have received a copy of the GNU Lesser General Public License
+    along with this program; if not, write to the Free Software
+    Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+    02110-1301  USA
+
+ */
+
+/*
+
+    These files are distributed with VIPS - http://www.vips.ecs.soton.ac.uk
+
+ */
+
+/*
+#define DEBUG_VERBOSE
+#define DEBUG
+ */
+
+#ifdef HAVE_CONFIG_H
+#include <config.h>
+#endif /*HAVE_CONFIG_H*/
+#include <vips/intl.h>
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include <vips/vips.h>
+
+#include "pforeign.h"
+
```

Comment:
I think macOS CI  fails due to the missing `quantise.h` include.
```suggestion
#include "quantise.h"

```

Reply:
Ooop, I rebased on master but didn't check it still built cleanly, argh. Thanks!